### ISSUE TITLE
Upgrade to Rust 1.0.0!

### ISSUE THREAD

**Issue by blast-hardcheese** [tinyplasticgreyknight/BANCNova#11]
Current version of Rust has many changes from when this was written. Now that the API has largely stabilized, it would be helpful to use it.

**Comment by tinyplasticgreyknight** [comment]
Yes; I haven't time to do it right now though.  Luckily there are maybe three people in the world who have any interest in BANCStar whatsoever, never mind actually _using_ it. ;-)

**Comment by blast-hardcheese** [comment]
@tinyplasticgreyknight Just for kicks, what version of Rust would this compile with? I found your repo from Esolangs, wanted to mess around with BANCStar a little this weekend.

**Comment by tinyplasticgreyknight** [comment]
Oof, I don't even know.  It was around the rustc-0.1 era, I think?

Some notes if you're planning to play around:
- This tool only deals with an assembly syntax, not execution.
- In fact, I don't think there are any available interpreters currently out there, unless @jloughry has managed to dig something up.
- If you glance through https://github.com/tinyplasticgreyknight/BANCNova/blob/master/doc/assembly.md, you'll notice a whole lot of "not very well understood" and "a mystery" comments.  The language is still a subject of archaeological research!
- This assembly tool itself has quite a few bugs and missing features, so I don't know how useful it will be yet.

**Comment by SwissKid** [comment]
Newly available interpreter (partially complete) https://github.com/SwissKid/bancstar-python
Because of a CTF at LayerOne security conference.

**Comment by tinyplasticgreyknight** [comment]
Interesting!  Did somebody dig up a specification?  Do you have a link?

**Comment by SwissKid** [comment]
Grabbed the spec from esolang. Its not following all the specs yet.

**Comment by tinyplasticgreyknight** [comment]
Ah; if you check the revision history on that article, you might notice a familiar username. ;-)

What exactly was happening at the conference you mention?  Did the topic of BANCStar come up?

**Comment by SwissKid** [comment]
You can see the SCN and PFL files I was given, that was the total of the
challenge.... they wouldn't even tell me it was bancstar!
